### DISCUSSION TITLE
[Feature] Private SMS Vault

### DISCUSSION THREAD

**Discussion by mojo-jojo-7** [FossifyOrg/General-Discussion#506] [Ideas]
**Private SMS Vault - What is it?**
A private database for storing messages received, into which messages can be moved. Consider the "Archive" feature, except that messages moved here will be deleted from the phone's SMS database and be only accessible inside the app. You could optionally encrypt this database, but that is secondary.

**What does this solve?**
A lot of applications, especially banking and finance related ones, require the "View and Send SMS" permission to run. I don't want to share all my messages with them. Having a private vault for SMS means I can now move all my SMS to this vault before granting SMS permissions to the app.

**Why not just backup/export messages and delete them manually?**
Because if, for some reason, I want to look at any of the backed up messages, it's a hassle. This new feature would offer a lot of convenience.

**"Yeah, but once you grant the SMS permission, the app will be able to see all new incoming messages anyway"**
Personally, I only need to use such apps once in a while, so I don't mind clearing the permissions manually after I'm done using them. Or, an app like [Bouncer](https://www.androidauthority.com/bouncer-android-app-905059/) can even automate this!

I hope you understood the use case here. Feel free to ask if anything is unclear. Finally, @naveensingh thanks a lot for keeping this suite of apps alive after the SMT fiasco.

**Comment by Aga-C** [comment]
It's already requested: https://github.com/FossifyOrg/Messages/issues/48.

**Comment by mojo-jojo-7** [comment]
Not quite. #48 is about hiding the chats behind a lock, my request is about removing the messages from the phone's SMS database and moving to a vault that only this app has access to. The first prevents other people from being able to access SMS, the latter prevents other apps on the phone being able to access SMS
